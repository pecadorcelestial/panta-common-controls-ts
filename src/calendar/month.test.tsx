
//RRRR  EEEEE  AAA   CCCC TTTTT
//R   R E     A   A C       T
//RRRR  EEE   AAAAA C       T
//R   R E     A   A C       T
//R   R EEEEE A   A  CCCC   T

import React from 'react';
import renderer from 'react-test-renderer';

//EEEEE N   N ZZZZZ Y   Y M   M EEEEE
//E     NN  N    Z   Y Y  MM MM E
//EEE   N N N   Z     Y   M M M EEE
//E     N  NN  Z      Y   M   M E
//EEEEE N   N ZZZZZ  YYY  M   M EEEEE

import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

enzyme.configure({ adapter: new Adapter() });

//JJJJJ EEEEE  SSSS TTTTT
//    J E     S       T
//    J EEE    SSS    T
//J   J E         S   T
// JJJ  EEEEE SSSS    T

//NOTA: Éste módulo ayuda a que los errores en componentes estilizados se vean de manera correcta:
//https://medium.com/styled-components/effective-testing-for-styled-components-67982eb7d42b
import 'jest-enzyme';
import 'jest-styled-components';

// CCCC  OOO  M   M PPPP   OOO  N   N EEEEE N   N TTTTT EEEEE  SSSS
//C     O   O MM MM P   P O   O NN  N E     NN  N   T   E     S
//C     O   O M M M PPPP  O   O N N N EEE   N N N   T   EEE    SSS
//C     O   O M   M P     O   O N  NN E     N  NN   T   E         S
// CCCC  OOO  M   M P      OOO  N   N EEEEE N   N   T   EEEEE SSSS

import Month from './month';

//Snapshot #1.
describe('[SNAPSHOT][Componentes][Common][Month] - Mes.', () => {
	//Tipo: Texto.
	it('Debe pintar el componente con los siguientes modificadores: (Tema: default)(Fecha: 15/08/1981).', () => {
		//Funciones 'dummy'.
		const handleMonthOnClick = event => {
			//Nada que hacer aquí.
        };
        const handleOnChange = date => {
			//Nada que hacer aquí.
        };
        const handleInnerOnChange = date => {
			//Nada que hacer aquí.
        };
        //Variables.
        let selectedDate = new Date('08/15/1981');
        let minDate = new Date('08/10/1981');
		//Propiedades.
        const monthProps = {
            //Obligatorios.
            selectedDate,
            innerDate: selectedDate,
            //Opcionales.
            language: 'es-MX',
            minDate,
            //Funciones.
            monthOnClick: handleMonthOnClick,
            onChange: handleOnChange,
            onInnerChange: handleInnerOnChange
        };
        //Se crea el componente.
        
        //N   N  OOO  RRRR  M   M  AAA  L
        //NN  N O   O R   R MM MM A   A L
        //N N N O   O RRRR  M M M AAAAA L
        //N  NN O   O R   R M   M A   A L
        //N   N  OOO  R   R M   M A   A LLLLL

        //Snapshot inicial.
        const component = renderer.create(<Month {...monthProps}/>).toJSON();
		expect(component).toMatchSnapshot();
	});
});

//Flujo #1.
describe('[FLUJO][Componentes][Common][Month] - Mes.', () => {
	//Funciones 'dummy'.
	const handleMonthOnClick = jest.fn(event => {
        //Nada que hacer aquí.
	});
	const handleOnChange = jest.fn(date => {
		//Nada que hacer aquí.
	});
	const handleInnerOnChange = jest.fn(date => {
		//Nada que hacer aquí.
	});
    //Variables.
    let selectedDate = new Date('08/15/1981');
    let minDate = new Date('08/10/1981');
    //Propiedades.
    const monthProps = {
        //Obligatorios.
        selectedDate,
        innerDate: selectedDate,
        //Opcionales.
        language: 'es-MX',
        minDate,
        //Funciones.
        monthOnClick: handleMonthOnClick,
        onChange: handleOnChange,
        onInnerChange: handleInnerOnChange
    };
	//Se crea el componente.
	//NOTA: Al utilizar la función 'mount' se detona las siguientes funciones: constructor, componentDidMount y render.
    const component = enzyme.mount(<Month {...monthProps}/>);

    //M   M  OOO  N   N TTTTT H   H  OOO  N   N  CCCC L     IIIII  CCCC K   K
    //MM MM O   O NN  N   T   H   H O   O NN  N C     L       I   C     K  K
    //M M M O   O N N N   T   HHHHH O   O N N N C     L       I   C     KKK
    //M   M O   O N  NN   T   H   H O   O N  NN C     L       I   C     K  K
    //M   M  OOO  N   N   T   H   H  OOO  N   N  CCCC LLLLL IIIII  CCCC K   K

    let monthButton = component.find('button#btn-month-year');
    it('Se revisa que exista 1 objeto tipo "button" con identificador "btn-month-year".', () => {
        expect(monthButton).toHaveLength(1);
    });
	it('Debe mandar llamar la función "handleMonthOnClick" que se pasa como propiedad.', () => {
		//Simulación.
        //monthButton.simulate('click');
        monthButton.prop('onClick')();
		//Expectativa.
		expect(handleMonthOnClick).toHaveBeenCalled();
	});

	// OOO  N   N  CCCC H   H  AAA  N   N  GGGG EEEEE
	//O   O NN  N C     H   H A   A NN  N G     E
	//O   O N N N C     HHHHH AAAAA N N N G  GG EEE
	//O   O N  NN C     H   H A   A N  NN G   G E
	// OOO  N   N  CCCC H   H A   A N   N  GGGG EEEEE

    let dayButtons = component.findWhere(node => {
        let regEx = /day-\d{1,2}-\d{1,2}/gi;
        return node.key() && node.key().match(regEx);
    });
    let bdButton = component.findWhere(node => node.key() && node.key() === 'day-15-20');
    it('Se revisa que existan 42 objetos tipo "button" con displayName "Day".', () => {
        expect(dayButtons).toHaveLength(42);
        expect(bdButton).toHaveLength(1);
    });
	it('Debe mandar llamar la función "handleOnChange" que se pasa como propiedad.', () => {
        //Simulación.
        bdButton.prop('onClick')();
		//Expectativas.
		expect(handleOnChange).toHaveBeenCalled();
    });

    //IIIII N   N N   N EEEEE RRRR   OOO  N   N  CCCC H   H  AAA  N   N  GGGG EEEEE
    //  I   NN  N NN  N E     R   R O   O NN  N C     H   H A   A NN  N G     E
    //  I   N N N N N N EEE   RRRR  O   O N N N C     HHHHH AAAAA N N N G  GG EEE
    //  I   N  NN N  NN E     R   R O   O N  NN C     H   H A   A N  NN G   G E
    //IIIII N   N N   N EEEEE R   R  OOO  N   N  CCCC H   H A   A N   N  GGGG EEEE

    let previousMonthButton = component.find('button#btn-previous-month');
    let nextMonthButton = component.find('button#btn-next-month');
    it('Se revisa que exista 1 objeto tipo "button" con identificador "btn-previous-month".', () => {
        expect(previousMonthButton).toHaveLength(1);
    });
    it('Se revisa que exista 1 objeto tipo "button" con identificador "btn-next-month".', () => {
        expect(nextMonthButton).toHaveLength(1);
    });
	it('Debe mandar llamar la función "handleMonthOnClick" que se pasa como propiedad.', () => {
		//Simulación.
        previousMonthButton.prop('onClick')();
		//Expectativa.
		expect(handleInnerOnChange).toHaveBeenCalled();
	});
	it('Debe mandar llamar la función "handleMonthOnClick" que se pasa como propiedad.', () => {
		//Simulación.
        nextMonthButton.prop('onClick')();
		//Expectativa.
		expect(handleInnerOnChange).toHaveBeenCalled();
	});
});
