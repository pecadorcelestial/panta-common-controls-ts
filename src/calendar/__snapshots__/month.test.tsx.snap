// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SNAPSHOT][Componentes][Common][Month] - Mes. Debe pintar el componente con los siguientes modificadores: (Tema: default)(Fecha: 15/08/1981). 1`] = `
.c0 {
  box-sizing: border-box;
  min-height: 195px;
  margin: 0px;
  overflow: hidden;
  padding: 0px;
  width: 185px;
}

.c1 {
  box-sizing: border-box;
  height: 60px;
  margin: 0px;
  padding: 5px;
  position: relative;
  text-align: center;
  width: 100%;
}

.c2 {
  background-color: transparent;
  border: none;
  border-radius: 5px;
  box-sizing: border-box;
  cursor: pointer;
  font-family: "Open Sans",sans-serif;
  font-size: 14px;
  font-weight: bold;
  height: 25px;
  margin: 0px 0px 5px 0px;
  padding: 2px 15px;
  width: auto;
}

.c2:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:focus {
  outline: none;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: auto;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin: 0px;
  padding: 0px;
  width: 100%;
}

.c6 {
  box-sizing: border-box;
  font-family: "Open Sans",sans-serif;
  font-size: 12px;
  font-weight: bold;
  height: 20px;
  margin: 0px;
  padidng: 2px;
  width: 20px;
}

.c7 {
  background-color: #FFF;
  box-sizing: border-box;
  height: auto;
  margin: 0px;
  padding: 5px;
  width: 100%;
}

.c3 {
  background-color: transparent;
  background-image: url("data:image/svg+xml;base64,PHN2ZyB2ZXJzaW9uPSIxLjEiIGlkPSJDYXBhXzEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IiB3aWR0aD0iMjkycHgiIGhlaWdodD0iMjkycHgiIHZpZXdCb3g9IjAgMCAyOTIuMzU5IDI5Mi4zNTkiIHN0eWxlPSJlbmFibGUtYmFja2dyb3VuZDpuZXcgMCAwIDI5Mi4zNTkgMjkyLjM1OTsgZmlsbDogd2hpdGU7IiB4bWw6c3BhY2U9InByZXNlcnZlIj48ZyB0cmFuc2Zvcm09InJvdGF0ZSgxODAsIDE0NiwgMTQ2KSI+PHBhdGggZD0iTTIyMi45NzksMTMzLjMzMUw5NS4wNzMsNS40MjRDOTEuNDU2LDEuODA3LDg3LjE3OCwwLDgyLjIyNiwwYy00Ljk1MiwwLTkuMjMzLDEuODA3LTEyLjg1LDUuNDI0Yy0zLjYxNywzLjYxNy01LjQyNCw3Ljg5OC01LjQyNCwxMi44NDd2MjU1LjgxM2MwLDQuOTQ4LDEuODA3LDkuMjMyLDUuNDI0LDEyLjg0N2MzLjYyMSwzLjYxNyw3LjkwMiw1LjQyOCwxMi44NSw1LjQyOGM0Ljk0OSwwLDkuMjMtMS44MTEsMTIuODQ3LTUuNDI4bDEyNy45MDYtMTI3LjkwN2MzLjYxNC0zLjYxMyw1LjQyOC03Ljg5Nyw1LjQyOC0xMi44NDdDMjI4LjQwNywxNDEuMjI5LDIyNi41OTQsMTM2Ljk0OCwyMjIuOTc5LDEzMy4zMzF6Ii8+PC9nPjwvc3ZnPg==");
  background-position: 1px center;
  background-repeat: no-repeat;
  background-size: 15px 15px;
  border: none;
  border-radius: 50%;
  cursor: pointer;
  height: 22px;
  left: 5px;
  position: absolute;
  top: 5px;
  width: 22px;
}

.c3:focus {
  outline: none;
}

.c4 {
  background-color: transparent;
  background-image: url("data:image/svg+xml;base64,PHN2ZyB2ZXJzaW9uPSIxLjEiIGlkPSJDYXBhXzEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IiB3aWR0aD0iMjkycHgiIGhlaWdodD0iMjkycHgiIHZpZXdCb3g9IjAgMCAyOTIuMzU5IDI5Mi4zNTkiIHN0eWxlPSJlbmFibGUtYmFja2dyb3VuZDpuZXcgMCAwIDI5Mi4zNTkgMjkyLjM1OTsgZmlsbDogd2hpdGU7IiB4bWw6c3BhY2U9InByZXNlcnZlIj48Zz48cGF0aCBkPSJNMjIyLjk3OSwxMzMuMzMxTDk1LjA3Myw1LjQyNEM5MS40NTYsMS44MDcsODcuMTc4LDAsODIuMjI2LDBjLTQuOTUyLDAtOS4yMzMsMS44MDctMTIuODUsNS40MjRjLTMuNjE3LDMuNjE3LTUuNDI0LDcuODk4LTUuNDI0LDEyLjg0N3YyNTUuODEzYzAsNC45NDgsMS44MDcsOS4yMzIsNS40MjQsMTIuODQ3YzMuNjIxLDMuNjE3LDcuOTAyLDUuNDI4LDEyLjg1LDUuNDI4YzQuOTQ5LDAsOS4yMy0xLjgxMSwxMi44NDctNS40MjhsMTI3LjkwNi0xMjcuOTA3YzMuNjE0LTMuNjEzLDUuNDI4LTcuODk3LDUuNDI4LTEyLjg0N0MyMjguNDA3LDE0MS4yMjksMjI2LjU5NCwxMzYuOTQ4LDIyMi45NzksMTMzLjMzMXoiLz48L2c+PC9zdmc+");
  background-position: right 1px center;
  background-repeat: no-repeat;
  background-size: 15px 15px;
  border: none;
  border-radius: 50%;
  cursor: pointer;
  height: 22px;
  position: absolute;
  right: 5px;
  top: 5px;
  width: 22px;
}

.c4:focus {
  outline: none;
}

.c8 {
  border: none;
  border-radius: 50%;
  box-sizing: border-box;
  cursor: pointer;
  font-family: "Open Sans",sans-serif;
  font-size: 14px;
  font-weight: bold;
  height: 25px;
  margin: 0px;
  padding: 0px;
  width: 25px;
  background-color: transparent;
  color: #BFBFBF;
}

.c8:hover {
  background-color: #BFBFBF;
  color: #FFF;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c8:focus {
  outline: none;
}

.c9 {
  border: none;
  border-radius: 50%;
  box-sizing: border-box;
  cursor: pointer;
  font-family: "Open Sans",sans-serif;
  font-size: 14px;
  font-weight: bold;
  height: 25px;
  margin: 0px;
  padding: 0px;
  width: 25px;
  background-color: transparent;
  color: #BFBFBF;
  cursor: default;
  pointer-events: none;
}

.c9:hover {
  background-color: transparent;
  color: #BFBFBF;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c9:focus {
  outline: none;
}

.c10 {
  border: none;
  border-radius: 50%;
  box-sizing: border-box;
  cursor: pointer;
  font-family: "Open Sans",sans-serif;
  font-size: 14px;
  font-weight: bold;
  height: 25px;
  margin: 0px;
  padding: 0px;
  width: 25px;
  background-color: transparent;
  color: #242424;
}

.c10:hover {
  background-color: #BFBFBF;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10:focus {
  outline: none;
}

.c11 {
  border: none;
  border-radius: 50%;
  box-sizing: border-box;
  cursor: pointer;
  font-family: "Open Sans",sans-serif;
  font-size: 14px;
  font-weight: bold;
  height: 25px;
  margin: 0px;
  padding: 0px;
  width: 25px;
  background-color: #1476FB;
  color: #FFF;
}

.c11:hover {
  background-color: #0960D3;
  color: #FFF;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c11:focus {
  outline: none;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <button
      className="c2"
      id="btn-month-year"
      onClick={[Function]}
      type="button"
    >
      M08 1981
    </button>
    <button
      className="c3"
      id="btn-previous-month"
      onClick={[Function]}
      type="button"
    />
    <button
      className="c4"
      id="btn-next-month"
      onClick={[Function]}
      type="button"
    />
    <div
      className="c5"
    >
      <label
        className="c6"
      >
        Do
      </label>
      <label
        className="c6"
      >
        Lu
      </label>
      <label
        className="c6"
      >
        Ma
      </label>
      <label
        className="c6"
      >
        Mi
      </label>
      <label
        className="c6"
      >
        Ju
      </label>
      <label
        className="c6"
      >
        Vi
      </label>
      <label
        className="c6"
      >
        Sa
      </label>
    </div>
  </div>
  <div
    className="c7"
  >
    <button
      className="c8"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      26
    </button>
    <button
      className="c8"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      27
    </button>
    <button
      className="c8"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      28
    </button>
    <button
      className="c8"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      29
    </button>
    <button
      className="c8"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      30
    </button>
    <button
      className="c8"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      31
    </button>
    <button
      className="c9"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="c9"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="c9"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="c9"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="c9"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="c9"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="c9"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="c9"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="c9"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      10
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      11
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      12
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      13
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      14
    </button>
    <button
      className="c11"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      15
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      16
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      17
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      18
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      19
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      20
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      21
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      22
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      23
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      24
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      25
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      26
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      27
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      28
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      29
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      30
    </button>
    <button
      className="c10"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      31
    </button>
    <button
      className="c8"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="c8"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="c8"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="c8"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="c8"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      5
    </button>
  </div>
</div>
`;
